diff --git a/src/weather.cpp b/src/weather.cpp
index a52d5052f6a7..b92863a186b7 100644
--- a/src/weather.cpp
+++ b/src/weather.cpp
@@ -889,7 +889,7 @@ bool warm_enough_to_plant( const tripoint_abs_omt &pos, const itype_id &it )
     // and now iterate a copy of the weather into the future to see if they'll be plantable then as well.
     time_duration one_growth_cycle = item( it ).get_plant_epoch( num_epochs );
     const weather_generator weather_gen = get_weather().get_cur_weather_gen();
-    for( int i = 0; i < num_epochs; i++ ) {
+    for( int i = 0; i < 0; i++ ) {
         // TODO: Replace epoch checks with data from a farmer's almanac
         check_date = check_date + one_growth_cycle;
         const w_point &w = weather_gen.get_weather( project_to<coords::ms>( pos ), check_date,
