diff --git a/src/activity_actor.cpp b/src/activity_actor.cpp
index 4c63a9368879..31c4f77a16be 100644
--- a/src/activity_actor.cpp
+++ b/src/activity_actor.cpp
@@ -4175,11 +4175,11 @@ void craft_activity_actor::do_turn( player_activity &act, Character &crafter )
     const double current_progress = craft.item_counter * base_total_moves / 10000000.0 +
                                     delta_progress;
     // Current progress as a percent of base_total_moves to 2 decimal places
-    craft.item_counter = std::round( current_progress / base_total_moves * 10000000.0 );
+    craft.item_counter = std::round( clamp( current_progress / base_total_moves * 10000000.0, 0.0, 10000000.0 ) );
     crafter.set_moves( 0 );

     // This is to ensure we don't over count skill steps
-    craft.item_counter = std::min( craft.item_counter, 10000000 );
+    craft.item_counter = clamp( craft.item_counter, 0, 10000000 );

     // This nominal craft time is also how many practice ticks to perform
     // spread out evenly across the actual duration.
