diff --git a/src/activity_actor.cpp b/src/activity_actor.cpp
index 387a37e4e057..45f6d346aa13 100644
--- a/src/activity_actor.cpp
+++ b/src/activity_actor.cpp
@@ -10499,7 +10499,7 @@ bool zone_sort_activity_actor::stage_think( player_activity &act, Character &you
     return true;
 }
 
-void zone_sort_activity_actor::stage_do( player_activity &act, Character &you )
+void zone_sort_activity_actor::stage_do( player_activity &, Character &you )
 {
     const map &here = get_map();
     const zone_manager &mgr = zone_manager::get_manager();
@@ -10551,11 +10551,13 @@ void zone_sort_activity_actor::stage_do( player_activity &act, Character &you )
         for( const tripoint_abs_ms &possible_dest : dest_set ) {
             const tripoint_bub_ms dest_bub = here.get_bub( possible_dest );
             // Routing will fail if we're already adjacent or at the destination. So we need to avoid checking it unless there's an actual route that needs to be checked.
+#if 0
             const bool is_adjacent_or_closer_to_dest = square_dist( you.pos_bub(), dest_bub ) <= 1;
             if( !is_adjacent_or_closer_to_dest &&
                 !zone_sorting::route_to_destination( you, act, dest_bub, stage ) ) {
                 continue; // Not a valid destination
             }
+#endif
             // This is a separate statement so we can have specific messaging if this failure state is reached.
             if( here.is_open_air( dest_bub ) ) {
                 you.add_msg_if_player( _( "You can't sort things into the air!" ) );
