From f31b8873d22b54f5076bcc692300f5ef84adbb8e Mon Sep 17 00:00:00 2001
From: RenechCDDA <84619419+RenechCDDA@users.noreply.github.com>
Date: Mon, 22 Dec 2025 16:02:03 -0500
Subject: [PATCH] Don't erase our coords_set, we're not *done*.

---
 src/activity_actor.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/activity_actor.cpp b/src/activity_actor.cpp
index 3a671797937e6..4682889b3d086 100644
--- a/src/activity_actor.cpp
+++ b/src/activity_actor.cpp
@@ -10208,7 +10208,6 @@ bool zone_sort_activity_actor::stage_think( player_activity &act, Character &you
     // iterate over zone positions and look for items to move
     for( const tripoint_abs_ms &src : src_sorted ) {
         placement = src;
-        coord_set.erase( src );
 
         const tripoint_bub_ms src_bub = here.get_bub( src );
         if( !here.inbounds( src_bub ) ) {
