diff --git a/data/core/external_options.json b/data/core/external_options.json
index 52b4263204f4..b83374f336c5 100644
--- a/data/core/external_options.json
+++ b/data/core/external_options.json
@@ -520,5 +520,40 @@
     "//": "The maximum number of overmaps that an intersection can deviate from its gridded position. Cannot be greater than or equal to row / 2 or column / 2, may cause bugs for > row / 4, column / 4.",
     "stype": "int",
     "value": 2
+  },
+  {
+    "type": "EXTERNAL_OPTION",
+    "name": "UNSTABLE_TRAIT_MUTATION_TIME",
+    "//": "The average hours between mutations with the unstable trait.",
+    "stype": "int",
+    "value": 48
+  },
+  {
+    "type": "EXTERNAL_OPTION",
+    "name": "CHAOTIC_TRAIT_MUTATION_TIME",
+    "//": "The average hours between mutations with the chaotic trait.",
+    "stype": "int",
+    "value": 12
+  },
+  {
+    "type": "EXTERNAL_OPTION",
+    "name": "BASE_MANA_REGEN_TIME",
+    "//": "How many hours it should take to regenerate all mana at base mana regen.",
+    "stype": "float",
+    "value": 8.0
+  },
+  {
+    "type": "EXTERNAL_OPTION",
+    "name": "BASE_MANA_REGEN_TIME",
+    "//": "How many hours it should take to regenerate all mana at base mana regen.",
+    "stype": "float",
+    "value": 8.0
+  },
+  {
+    "type": "EXTERNAL_OPTION",
+    "name": "CROP_GROWTH_TIME_SCALER",
+    "//": "A scaler for how long crops take to grow. 0.5 will make crops take half as much time to grow.",
+    "stype": "float",
+    "value": 1.0
   }
 ]
diff --git a/src/magic.cpp b/src/magic.cpp
index 27f9f39a42cf..2ece788f73c9 100644
--- a/src/magic.cpp
+++ b/src/magic.cpp
@@ -2477,7 +2477,8 @@ int known_magic::max_mana( const Character &guy ) const
 void known_magic::update_mana( const Character &guy, float turns )
 {
     // mana should replenish in 8 hours.
-    const double full_replenish = to_turns<double>( 8_hours );
+    const double full_replenish =
+        to_turns<double>( 1_hours ) * ::get_option<float>( "BASE_MANA_REGEN_TIME" );
     const double ratio = turns / full_replenish;
     mod_mana( guy, std::floor( ratio * std::max( 0.0,
                                guy.calculate_by_enchantment( static_cast<double>( max_mana(
diff --git a/src/map.cpp b/src/map.cpp
index 63660372f817..03b6971e2bc0 100644
--- a/src/map.cpp
+++ b/src/map.cpp
@@ -8611,7 +8611,7 @@ void map::grow_plant( const tripoint_bub_ms &p )
             target_stage = pair.first;
         } // Don't break the loop for the case where time has been rewound.
         // Advance to the time of the next stage for the next iteration.
-        time_to_grow_to_this_stage += pair.second;
+        time_to_grow_to_this_stage += pair.second * ::get_option<float>( "CROP_GROWTH_TIME_SCALER" );
     }
 
     const auto check_flag = []( const std::string & to_check ) {
diff --git a/src/suffer.cpp b/src/suffer.cpp
index f0c234fac297..b35c7f978ebd 100644
--- a/src/suffer.cpp
+++ b/src/suffer.cpp
@@ -1112,9 +1112,9 @@ void suffer::from_other_mutations( Character &you )
     }
 
     bool should_mutate = you.has_trait( trait_UNSTABLE ) && !you.has_trait( trait_CHAOTIC_BAD ) &&
-                         one_turn_in( 48_hours );
+                         one_turn_in( 1_hours * ::get_option<int>( "UNSTABLE_TRAIT_MUTATION_TIME" ) );
     should_mutate |= ( you.has_trait( trait_CHAOTIC ) || you.has_trait( trait_CHAOTIC_BAD ) ) &&
-                     one_turn_in( 12_hours );
+                     one_turn_in( 1_hours * ::get_option<int>( "CHAOTIC_TRAIT_MUTATION_TIME" ) );
     if( should_mutate ) {
         you.mutate();
     }
