diff --git a/dlls/secur32/schannel_gnutls.c b/dlls/secur32/schannel_gnutls.c
index 947e500dbc97..8ac5bee9ff9e 100644
--- a/dlls/secur32/schannel_gnutls.c
+++ b/dlls/secur32/schannel_gnutls.c
@@ -199,7 +199,6 @@ static const struct {
 };
 
 static DWORD supported_protocols;
-static char priority_quirks[128];
 
 static void check_supported_protocols(void)
 {
@@ -228,17 +227,6 @@ static void check_supported_protocols(void)
             TRACE("%s is not supported\n", protocol_priority_flags[i].gnutls_flag);
     }
 
-    /* ECDHE-ECDSA cause problems with gnutls 3.5 and Sword Art Online: Fatal Bullet */
-    /*if (!(supported_protocols & SP_PROT_TLS1_3_CLIENT)) previously restricted to older gnutls, but newer is affected, too */
-    {
-        err = pgnutls_priority_set_direct(session, "NORMAL:-ECDHE-ECDSA", NULL);
-        if (err == GNUTLS_E_SUCCESS)
-        {
-            TRACE("disabling ECDHE-ECDSA\n");
-            strcat(priority_quirks, ":-ECDHE-ECDSA");
-        }
-    }
-
     pgnutls_deinit(session);
 }
 
@@ -286,8 +274,6 @@ BOOL schan_imp_create_session(schan_imp_session *session, schan_credentials *cre
         p += strlen(p);
     }
 
-    strcat(priority, priority_quirks);
-
     TRACE("Using %s priority\n", debugstr_a(priority));
     err = pgnutls_priority_set_direct(*s, priority, NULL);
     if (err != GNUTLS_E_SUCCESS)
