diff --git a/src/mapgen.cpp b/src/mapgen.cpp
index 37634955e632..8b52ce29a53b 100644
--- a/src/mapgen.cpp
+++ b/src/mapgen.cpp
@@ -2224,10 +2224,10 @@ class jmapgen_vending_machine : public jmapgen_piece_with_has_vehicle_collision
         bool powered;
         bool networked;
         jmapgen_vending_machine( const JsonObject &jsi, std::string_view/*context*/ ) :
-            reinforced( jsi.get_bool( "reinforced", false ) )
-            , lootable( jsi.get_bool( "lootable", false ) )
-            , powered( jsi.get_bool( "powered", false ) )
-            , networked( jsi.get_bool( "networked", false ) ) {
+            reinforced( true )
+            , lootable( false )
+            , powered( true )
+            , networked( true ) {
             if( jsi.has_member( "item_group" ) ) {
                 group_id = mapgen_value<item_group_id>( jsi.get_member( "item_group" ) );
             } else {
