diff --git a/src/trade_ui.cpp b/src/trade_ui.cpp
index 0f1eb47db886..b1ca22af050d 100644
--- a/src/trade_ui.cpp
+++ b/src/trade_ui.cpp
@@ -129,7 +129,7 @@ trade_ui::trade_ui( party_t &you, npc &trader, currency_t cost, std::string titl
 
 {
     _panes[_you]->add_character_items( you );
-    _panes[_you]->add_nearby_items( 1 );
+    _panes[_you]->add_nearby_items( 8 );
     if( !trader.has_trait( trait_TRADE_BACKEND ) ) {
         _panes[_trader]->add_character_items( trader );
     }
