diff --git a/src/veh_interact.cpp b/src/veh_interact.cpp
index f67d7b170d58..231d459cce52 100644
--- a/src/veh_interact.cpp
+++ b/src/veh_interact.cpp
@@ -844,16 +844,15 @@ bool veh_interact::update_part_requirements( map &here )
 
     bool is_engine = sel_vpart_info->has_flag( "ENGINE" );
     //count current engines, some engines don't require higher skill
-    int engines = 0;
     int dif_eng = 0;
     if( is_engine && sel_vpart_info->has_flag( "E_HIGHER_SKILL" ) ) {
         for( const vpart_reference &vp : veh->get_avail_parts( "ENGINE" ) ) {
             if( vp.has_feature( "E_HIGHER_SKILL" ) ) {
-                engines++;
                 dif_eng = dif_eng / 2 + 8;
             }
         }
     }
+    dif_eng = std::min(dif_eng, 8);
 
     int dif_steering = 0;
     if( sel_vpart_info->has_flag( "STEERABLE" ) ) {
@@ -873,6 +872,8 @@ bool veh_interact::update_part_requirements( map &here )
         }
     }
 
+    dif_steering = std::min(dif_steering, 8);
+
     const requirement_data reqs = sel_vpart_info->install_requirements();
 
     avatar &player_character = get_avatar();
@@ -882,7 +883,7 @@ bool veh_interact::update_part_requirements( map &here )
 
     nmsg += _( "<color_white>Additional requirements:</color>\n" );
 
-    bool allow_more_eng = engines < 2 || player_character.has_trait( trait_DEBUG_HS );
+    bool allow_more_eng = true;
 
     if( dif_eng > 0 ) {
         if( !allow_more_eng || player_character.get_knowledge_level( skill_mechanics ) < dif_eng ) {
