diff --git a/src/iexamine.cpp b/src/iexamine.cpp
index 57b617f346d8..5cb0b0f79057 100644
--- a/src/iexamine.cpp
+++ b/src/iexamine.cpp
@@ -7531,7 +7531,7 @@ iexamine_functions iexamine_functions_from_string( const std::string &function_n
 void iexamine::practice_survival_while_foraging( Character &who )
 {
     ///\EFFECT_INT Intelligence caps survival skill gains from foraging
-    const int max_forage_skill = who.get_int() / 3 + 1;
+    const int max_forage_skill = who.get_int() / 2 + 1;
     ///\EFFECT_SURVIVAL decreases survival skill gain from foraging (NEGATIVE)
     const int max_exp = 2 * ( max_forage_skill - static_cast<int>( who.get_skill_level(
                                   skill_survival ) ) );
