diff --git a/src/port.h b/src/port.h
index ff1887136699..9c70ad55b70a 100644
--- a/src/port.h
+++ b/src/port.h
@@ -112,6 +116,7 @@ extern void free();

 /* Try the common case insensitive strcmp's before falling back to our own */
 #if HAVE_STRCASECMP
+# include <strings.h>
 # define cstrcmp   strcasecmp
 #else
 # if HAVE_STRICMP
@@ -215,6 +220,7 @@ extern char *sys_errlist[];
  */

 #if HAVE_SRANDOM
+# include <stdlib.h>
 # include <math.h>
 # define RAND()         (int)random()
 # define SRAND(seed)    srandom(seed)
diff --git a/src/tfconfig.h.in b/src/tfconfig.h.in
index 56c707657040..f3fbba3f6cc4 100644
--- a/src/tfconfig.h.in
+++ b/src/tfconfig.h.in
@@ -103,5 +103,9 @@
 #include "tfdefs.h"

 #undef NDEBUG
-#include <assert.h>

+#define _POSIX_C_SOURCE 200809L
+#define _XOPEN_SOURCE 600
+#define _DEFAULT_SOURCE 1
+
+#include <assert.h>
